In [2]:
import pandas as pd
import csv

In [4]:
df = pd.read_fwf('~/persistent_files/raw-data/data/rt_vehicles_DB_2018.txt')

FileNotFoundError: [Errno 2] No such file or directory: '/home/student/persistent_files/raw-data/data/rt_vehicles_DB_2018.txt'

In [11]:
df.head()

DATASOURCE;DAYOFSERVICE;VEHICLEID;DISTANCE;MINUTES;LASTUPDATE;NOTE
0  DB;23-NOV-18 00:00:00;3303848;286166;58849;04-...                
1  DB;23-NOV-18 00:00:00;3303847;259545;56828;04-...                
2  DB;28-FEB-18 00:00:00;2868329;103096;40967;08-...                
3  DB;28-FEB-18 00:00:00;2868330;147277;43599;08-...                
4  DB;28-FEB-18 00:00:00;2868331;224682;40447;08-...

In [12]:
df.to_csv('Vehicles.csv')

In [3]:
df = pd.read_csv('Vehicles.csv', keep_default_na=True, sep=',\s+', delimiter=';', skipinitialspace=True)

FileNotFoundError: [Errno 2] File Vehicles.csv does not exist: 'Vehicles.csv'

In [14]:
df.head()

,DATASOURCE        DAYOFSERVICE  VEHICLEID  DISTANCE  MINUTES  \
0        0,DB  23-NOV-18 00:00:00    3303848    286166    58849   
1        1,DB  23-NOV-18 00:00:00    3303847    259545    56828   
2        2,DB  28-FEB-18 00:00:00    2868329    103096    40967   
3        3,DB  28-FEB-18 00:00:00    2868330    147277    43599   
4        4,DB  28-FEB-18 00:00:00    2868331    224682    40447   

           LASTUPDATE  NOTE  
0  04-DEC-18 08:03:09   NaN  
1  04-DEC-18 08:03:09   NaN  
2  08-MAR-18 10:35:59   NaN  
3  08-MAR-18 10:35:59   NaN  
4  08-MAR-18 10:35:59   NaN

In [15]:
df.tail()

,DATASOURCE        DAYOFSERVICE  VEHICLEID  DISTANCE  MINUTES  \
272617   272617,DB  29-DEC-18 00:00:00    3393878    264237    62320   
272618   272618,DB  29-DEC-18 00:00:00    3394105    250335    52254   
272619   272619,DB  29-DEC-18 00:00:00    3394109    172539    44349   
272620   272620,DB  29-DEC-18 00:00:00    3394130    188057    38948   
272621   272621,DB  29-DEC-18 00:00:00    3394131    291697    63677   

                LASTUPDATE  NOTE  
272617  16-JAN-19 18:00:42   NaN  
272618  16-JAN-19 18:00:42   NaN  
272619  16-JAN-19 18:00:42   NaN  
272620  16-JAN-19 18:00:42   NaN  
272621  16-JAN-19 18:00:42   NaN

In [16]:
df.dtypes

,DATASOURCE      object
DAYOFSERVICE     object
VEHICLEID         int64
DISTANCE          int64
MINUTES           int64
LASTUPDATE       object
NOTE            float64
dtype: object

In [17]:
df.shape

(272622, 7)

In [19]:
print('Number of duplicate (excluding first) rows in the table is: ', df.duplicated().sum())

# Check for duplicate rows. 
# Use "keep=False" to mark all duplicates as true, including the original rows that were duplicated.
print('Number of duplicate rows (including first) in the table is:', df[df.duplicated(keep=False)].shape[0])

Number of duplicate (excluding first) rows in the table is:  0
Number of duplicate rows (including first) in the table is: 0


In [18]:
df.nunique()

,DATASOURCE     272622
DAYOFSERVICE       360
VEHICLEID         1152
DISTANCE        170498
MINUTES          57523
LASTUPDATE         360
NOTE                 0
dtype: int64

So I can conclude here that we have 272622 entries
1152 buses
Represents 360 days worth of data
The distance travelled by the vehicle in that day.
The time (in minutes) worked by the vehicle that day.

What's useful here?
There are no notes and DATASOURCE is a constant column so I can remove both of them, LASTUPDATE doesn't add much to my knowledge so that can be dropped.
DAY OF SERVICE - continuous feature - need datetime
VEHICLE ID - categorical feature.
DISTANCE - continuous feature
MINUTES - continuous feature


In [ ]:
df['DAYOFSERVICE'] = df['DAYOFSERVICE'].astype('datetime64[ns]')
df['VEHICLEID'] = df['VEHICLEID'].astype('category')
del df['NOTE']
del df['LASTUPDATE']
del df[',DATASOURCE']


In [21]:
df.describe()

VEHICLEID       DISTANCE        MINUTES  NOTE
count  2.726220e+05  272622.000000  272622.000000   0.0
mean   1.928436e+06  177486.328466   45597.643220   NaN
std    8.104407e+05   71334.432383   14881.389868   NaN
min    1.000110e+06       0.000000  -70145.000000   NaN
25%    1.000948e+06  126765.000000   39380.000000   NaN
50%    2.172252e+06  177420.000000   46994.500000   NaN
75%    2.693217e+06  227126.250000   57911.000000   NaN
max    3.394131e+06  473864.000000   76651.000000   NaN